### PR TITLE
Some additional improvements to add more type-safety

### PR DESCRIPTION
**Pull request by frederikprijck**
### Changes

This PR adds a couple of additional type-safety measurements. I also bumped prettier to ensure we support pretty'ing modern TypeScript syntax.

Also dropped our own custom `RequestOptions`, and used Fetch' built-in `RequestInit`, as we do not expose that there's no real need to have our own type here, especially given it limits the fetch API internally.

This also adds a custom test environment to ensure the tests run in an environment that supports native Fetch, which isn't supported in Jest yet, see https://github.com/jsdom/jsdom/issues/1724.

This means we needed to regenerate some snapshots because some internals have changed.

### Checklist

- [x] I have read the [Auth0 general contribution guidelines](https://github.com/auth0/open-source-template/blob/master/GENERAL-CONTRIBUTING.md)
- [x] All existing and new tests complete without errors
- [x] All active GitHub checks have passed

### CODE REVIEW

File: src/utils/fetchWithTimeout.ts [hunk 1]
```diff
@@ -21,9 +21,12 @@ function makeTimeout(timeoutMs: number) {
   };
 }
 
+/**
+ * @private
+ */
 export function fetchWithTimeout(
   url: string,
-  options: RequestOptions,
+  options: RequestInit,
```
Comment (frederikprijck) [inline src/utils/fetchWithTimeout.ts:29]: By dropping our own RequestOptions, which is a limited version of Fetch' RequestInit, we can not use fetchWithTimeout for more than whats defined on our own RequestOptions. Even though that might not cause issues, it felt like it was unneccesary.

By dropping this, this method does one thing only: `wrap a timeout around fetch`, and no longer limits the options to provide to fetch.

File: src/utils/camel.ts [hunk 2]
```diff
@@ -1,13 +1,19 @@
-export function snakeToCamel(str: string): string {
+/**
+ * @private
+ **/
```
Comment (frederikprijck) [inline src/utils/camel.ts:3]: oesn't matter yet, but added @Private while I was at it. This ensures it wont show up in TSDocs.

File: src/networking/index.ts [hunk 3]
```diff
@@ -74,22 +74,24 @@ export default class Client {
...
+    headers.set('Auth0-Client', this._encodedTelemetry());
+
+    const options: RequestInit = {
+      method,
+      headers,
     };
 
     if (this.bearer) {
-      options.headers.Authorization = this.bearer;
+      headers.set('Authorization', this.bearer);
```
Comment (frederikprijck) [inline src/networking/index.ts:94]: Changed how we set the headers as TypeScript doesnt allow you to set the headers the way we did before.

Instead, used the recommend way to initialize Headers, then set the values.

File: src/networking/index.ts [hunk 4]
```diff
@@ -49,16 +49,16 @@ export default class Client {
...
-  patch(path: string, body: unknown) {
-    return this.request('PATCH', this.url(path), body);
+  patch<TData = unknown, TBody = unknown>(path: string, body: TBody) {
+    return this.request<TData, TBody>('PATCH', this.url(path), body);
   }
 
-  get(path: string, query?: unknown) {
-    return this.request('GET', this.url(path, query));
+  get<TData = unknown>(path: string, query?: unknown) {
+    return this.request<TData>('GET', this.url(path, query));
```
Comment (frederikprijck) [inline src/networking/index.ts:61]: Ensures to pass the Response type, and payload where needed to improve type safety.

**Review by poovamraj — APPROVED** [review]
Thanks for these great improvements @frederikprijck

**Pull request merged**